python/lsst/pipe/tasks/calibrate.py

```diff
@@ -21,151 +21,80 @@
-        target = MeasureApCorrTask,
-        doc = "subtask to measure aperture corrections",
-    )
-    astrometry = pexConfig.ConfigurableField(
-        target = AstrometryTask,
-        doc = "fit WCS of exposure",
+        default = True,
+        doc = "Raise an exception if photoCal fails? Ignored if doPhotoCal false.",
     )
-    photocal = pexConfig.ConfigurableField(
+    photoCal = pexConfig.ConfigurableField(
         target = PhotoCalTask,
-        doc = "peform photometric calibration",
+        doc = "Perform photometric calibration",
+    )
+    icSourceFieldsToCopy = pexConfig.ListField(
+        dtype = str,
+        default = ("calib_psfCandidate", "calib_psfUsed", "calib_psfReserved"),
+        doc = "Fields to copy from the icSource catalog to the output catalog for matching sources. "
+        "Missing fields are logged (at debug level) and ignored. "
+        "If detectAndMeasure.doMeasureApCorr is True and detectAndMeasure cannot determine its own "
+        "suitable candidates, then this list must include "
+        "config.detectAndMeasure.measureApCorr.inputFilterFlag."
     )
-
-    def validate(self):
-        pexConfig.Config.validate(self)
-        if self.doPhotoCal and not self.doAstrometry:
-            raise ValueError("Cannot do photometric calibration without doing astrometric matching")
-        if self.requireAstrometry and not self.doAstrometry:
-            raise ValueError("Astrometric solution required, but not activated")
-        if self.requirePhotoCal and not self.doPhotoCal:
-            raise ValueError("Photometric calibration required, but not activated")
-        if self.doMeasureApCorr and self.measureApCorr.inputFilterFlag == "calib_psfUsed" and not self.doPsf:
-            raise ValueError("Cannot measure aperture correction with inputFilterFlag=calib_psfUsed"
-                " unless doPsf is also True")
-        if self.measurement.doApplyApCorr.startswith("yes") and not self.doMeasureApCorr:
-            raise ValueError("Cannot set measurement.doApplyApCorr to 'yes...'"
-                " unless doMeasureApCorr is True")
 
     def setDefaults(self):
-        self.detection.includeThresholdMultiplier = 10.0
-        # Because we don't deblend in CalibrateTask, we want to minimize merge detections caused by
-        # growing the footprints.
-        self.detection.returnOriginalFootprints = True
-        self.initialMeasurement.algorithms.names -= ["base_ClassificationExtendedness"]
-        self.initialMeasurement.doApplyApCorr = "no" # no aperture correction data yet
-        self.measurement.doApplyApCorr = "yes"
-        initflags = [x for x in self.measurePsf.starSelector["catalog"].badStarPixelFlags]
-        self.measurePsf.starSelector["catalog"].badStarPixelFlags.extend(initflags)
-        self.background.binSize = 1024
+        pexConfig.Config.setDefaults(self)
+        # aperture correction should already be measured
+        self.detectAndMeasure.doMeasureApCorr = False
```

Comment:
Why is measuring the aperture correction part of `detectAndMeasure`?  That it has to be disabled here may indicate that the model isn't quite right.  Maybe it shouldn't be part of `detectAndMeasure`, or maybe it should be an API parameter rather than a config parameter.